tests/utils.py

```diff
@@ -160,7 +160,7 @@ def add_new_block(manager, advance_clock=None, *, parent_block_hash=None, data=b
     if weight is not None:
         block.weight = weight
     block.resolve()
-    block.verify()
+    block.validate_full()
```

Comment:
Isn't `verify()` and `validate_full()` the same? If not, where do they differ?

Reply:
`verify` does not write on the database, `validate_full` will update (and save?) the `metadata.validation` state

Reply:
Also, `validate_full` is _supposed_ to skip some verifications if `validate_basic` has been run before, and also `validate_full` can `skip_block_weight_verification`, while `verify` cannot. ("_supposed_" but it calls `.verify()` internally instead of only running a `verify_post_basic` or something like that)

I considered renaming the `verify` family of methods like we talked but it's too widespread and will result in a very large (although simple) refactor.

I also tried unifying `verify` and `validate_full` but because `validate_full` makes some assumptions about the metadata and also runs `verify_basic` before `verify`.

Either way, I'm moving this to an issue and so I can merge this PR. I completely agree we should improve how this is being handled to minimize any confusion.